<launch>

    <!-- Experiment parameters -->

        <!-- average_entropy, occlusion_aware, unknown_voxels, random, hidden_voxel, hidden_entropy, voxel_count-->
        <arg name="experiment_name"     default="fish_1"/> 
        <arg name="time_step"           default="5"/>
        <arg name="initial_view"        default="camera_front"/>

    <!-- Map parameters -->

    <arg name="resolution"      default="0.005"/>
	<arg name="depth_levels"    default="16"/>
	<arg name="multithreaded"   default="false"/>
    <arg name="insert_depth"    default="0"/>
    <arg name="insert_n"        default="0"/>
    <arg name="max_range"       default="1.0"/>

    <node pkg="dynamic_view_planning" type="ig_server_node" name="ig_server_node" output="screen"/>

    <node pkg="dynamic_view_planning" type="view_planning_node" name="view_planning_node_occlusion_aware" output="screen">
        
        <param name="ig_formulation"    value="occlusion_aware" />
        <param name="initial_view"      value="$(arg initial_view)"/>
        <param name="force_movement"    value="false"/>
        
        <param name="experiment_name"   value="$(arg experiment_name)"/>
        <param name="time_step"         value="$(arg time_step)"/>
        <param name="multithreaded"     value="$(arg multithreaded)"/>	
        <param name="resolution"        value="$(arg resolution)"/>
		<param name="depth_levels"      value="$(arg depth_levels)"/>
        <param name="insert_depth"      value="$(arg insert_depth)"/>
        <param name="insert_n"          value="$(arg insert_n)"/>
        <param name="max_range"         value="$(arg max_range)"/>

    </node>

    <node pkg="dynamic_view_planning" type="view_planning_node" name="view_planning_node_occlusion_aware_FM" output="screen">
        
        <param name="ig_formulation"    value="occlusion_aware" />
        <param name="initial_view"      value="$(arg initial_view)"/>
        <param name="force_movement"    value="true"/>
        
        <param name="experiment_name"   value="$(arg experiment_name)"/>
        <param name="time_step"         value="$(arg time_step)"/>
        <param name="multithreaded"     value="$(arg multithreaded)"/>	
        <param name="resolution"        value="$(arg resolution)"/>
		<param name="depth_levels"      value="$(arg depth_levels)"/>
        <param name="insert_depth"      value="$(arg insert_depth)"/>
        <param name="insert_n"          value="$(arg insert_n)"/>
        <param name="max_range"         value="$(arg max_range)"/>

    </node>

    <node pkg="dynamic_view_planning" type="view_planning_node" name="view_planning_node_average_entropy" output="screen">
        
        <param name="ig_formulation"    value="average_entropy" />
        <param name="initial_view"      value="$(arg initial_view)"/>
        <param name="force_movement"    value="false"/>
        
        <param name="experiment_name"   value="$(arg experiment_name)"/>
        <param name="time_step"         value="$(arg time_step)"/>
        <param name="multithreaded"     value="$(arg multithreaded)"/>	
        <param name="resolution"        value="$(arg resolution)"/>
		<param name="depth_levels"      value="$(arg depth_levels)"/>
        <param name="insert_depth"      value="$(arg insert_depth)"/>
        <param name="insert_n"          value="$(arg insert_n)"/>
        <param name="max_range"         value="$(arg max_range)"/>

    </node>

    <node pkg="dynamic_view_planning" type="view_planning_node" name="view_planning_node_unknown_voxels" output="screen">
        
        <param name="ig_formulation"    value="unknown_voxels" />
        <param name="initial_view"      value="$(arg initial_view)"/>
        <param name="force_movement"    value="false"/>
        
        <param name="experiment_name"   value="$(arg experiment_name)"/>
        <param name="time_step"         value="$(arg time_step)"/>
        <param name="multithreaded"     value="$(arg multithreaded)"/>	
        <param name="resolution"        value="$(arg resolution)"/>
		<param name="depth_levels"      value="$(arg depth_levels)"/>
        <param name="insert_depth"      value="$(arg insert_depth)"/>
        <param name="insert_n"          value="$(arg insert_n)"/>
        <param name="max_range"         value="$(arg max_range)"/>

    </node>



</launch>